### PR TITLE
Refactor to improve stability and reduce warnings; simplify MainView code

### LINKED ISSUE #66: The empty screen doesn't communicate scanning?
The current empty screen, shown when no devices are found yet, doesn't really communicate that it's scanning.
screenshot from 2018-11-13 16 43 42

Shouldn't we indicate we're stilling scanning, include a spinner?

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/66114160-the-empty-screen-doesn-t-communicate-scanning?utm_campaign=plugin&utm_content=tracker%2F59778156&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F59778156&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by jeremypw**
Fixes #66 

* Use property notifications more
* Transfer some business code from View to Manager
* Do not access adapter list in View
* Lose unnecessary main_adapter member in View
* Check global state in Idle loop not signal handler
* Reschedule various operations
* Discovery spinner in separate grid so always appears when discovering (also silences critical warning).

UI for discovery spinner only a suggestion.

### CODE REVIEW

**Comment by cassidyjames** [comment]
@jeremypw can you check the conflicts here?

**Comment by jeremypw** [comment]
@cassidyjames conflicts resolved

File: src/Services/Manager.vala [hunk 1]
```diff
@@ -170,6 +175,16 @@ public class Bluetooth.Services.ObjectManager : Object {
         }
     }
 
+    public string? get_name () {
```
Comment (tintou) [inline src/Services/Manager.vala]:
```suggestion
    public unowned string? get_name () {
```

File: src/MainView.vala [hunk 2]
```diff
@@ -184,38 +206,10 @@ public class Bluetooth.MainView : Granite.SimpleSettingsPage {
...
-
-        if (discovering != null) {
-            spinner.active = adapter.discovering;
-        }
-    }
-
-    private void update_description (string? name, bool discoverable, bool powered) {
-        if (discoverable && powered) {
+    private void update_description () {
+        string? name = manager.get_name ();
```
Comment (tintou) [inline src/MainView.vala]:
```suggestion
        unowned string? name = manager.get_name ();
```

File: src/Services/Manager.vala (hunk 1 above)
Comment (jeremypw) [reply src/Services/Manager.vala]: I get a compile error if I do this:
```
Manager.vala:183.17-183.64: error: Return value transfers ownership but method return type hasn't been declared to transfer ownership

```

Comment (tintou) [reply src/Services/Manager.vala]: Ah yeah haven't saw that it was owned here, so everything is fine

**Pull request merged**